ambry-router/src/main/java/com/github/ambry/router/UndeleteOperation.java

```diff
@@ -195,6 +194,7 @@ void handleResponse(ResponseInfo responseInfo, UndeleteResponse undeleteResponse
               // This is first successful response.
               lifeVersion = undeleteResponse.getLifeVersion();
               firstResponseReplicaId = replica;
+              operationTracker.onResponse(replica, TrackedRequestFinalState.SUCCESS);
```

Comment:
How is the life version field in the error UndeleteResponse used?

Reply:
when the error is Blob_Already_Undelete, we treat it as a successful operation, just like in DeleteOperation, when the error is Blob_Already_Deleted, it's considered as a successful operation. The lifeVersion will be used to compare with other ambry servers' lifeVersions.